tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,210 @@
+
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have the expected entries in vmi status", func() {
+			expectedInterfaces := []kvirtv1.VirtualMachineInstanceNetworkInterface{
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        originalMac,
+					Name:       defaultInterfaceName,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceDomainAndGA,
+					InterfaceName: "eth1",
+					MAC:           secondaryLinuxBridgeInterface1.MacAddress,
+					Name:          secondaryLinuxBridgeInterface1.Name,
+				},
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        secondaryLinuxBridgeInterface2.MacAddress,
+					Name:       secondaryLinuxBridgeInterface2.Name,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "eth0",
+					MAC:           updatedMac,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "dummy0",
+					MAC:           dummyInterfaceMac,
+				},
+			}
+
+			var vmiObj *kvirtv1.VirtualMachineInstance
+			Eventually(func() bool {
+				var err error
+				vmiObj, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				return dummyInterfaceExists(vmiObj)
+			}, 140*time.Second, 2*time.Second).Should(Equal(true))
+
+			Expect(len(vmiObj.Status.Interfaces)).To(Equal(len(expectedInterfaces)))
+
+			canonizeInterfaceStatus(vmiObj)
+			for i := range vmiObj.Status.Interfaces {
+				expectedInterfaces[i].IP = vmiObj.Status.Interfaces[i].IP
+				expectedInterfaces[i].IPs = vmiObj.Status.Interfaces[i].IPs
+			}
+
+			Expect(reflect.DeepEqual(expectedInterfaces, vmiObj.Status.Interfaces)).To(BeTrue(),
+				fmt.Sprintf("interfaces doesn't match\n%v\n%v", expectedInterfaces, vmiObj.Status.Interfaces))
+		})
+	})
+})
+
+func newNetworkAttachmentDefinition(networkName string) *nadv1.NetworkAttachmentDefinition {
+	config := fmt.Sprintf(`{"cniVersion": "0.3.1", "name": "%s", "type": "cnv-bridge", "bridge": "%s"}`, networkName, networkName)
+	return &nadv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: networkName,
+		},
+		Spec: nadv1.NetworkAttachmentDefinitionSpec{Config: config},
+	}
+}
+
+// dummyInterfaceExists waits for least the dummy interface to be reported by guest agent,
+// which means there was a guest-agent report finally, and then we can compare the rest expected info.
+func dummyInterfaceExists(vmi *kvirtv1.VirtualMachineInstance) bool {
+	for i := range vmi.Status.Interfaces {
+		if vmi.Status.Interfaces[i].InterfaceName == "dummy0" {
+			return true
+		}
+	}
+	return false
+}
+
+// canonizeInterfaceStatus orders "guest-agent" only records in predictable order
+func canonizeInterfaceStatus(vmi *kvirtv1.VirtualMachineInstance) {
```

Comment:
Can you please explain why this can happen? I would expect the order to be consistent.
In any case, the name is not reflecting what it does really... you just order here the list.

Also, please pass in only what is needed (`Interfaces`) and make this immutable (i.e. return a new list).

Reply:
> Can you please explain why this can happen? I would expect the order to be consistent.

Guest agent report them in a random order, even on the same run